Choose Your Own Emoji.md

```diff
@@ -98,13 +98,32 @@ Html.Events.onClick (Update.SetSelectedKey emoji)
 
 When this element is clicked, a `Msg` of `Update.SetSelectedKey String` will be triggered, which will eventually be consumed by our application.
 
-This is a good time to talk about the strange value of the `Msg` union type that seems to have the type `String` attached to the end of it. We've seen this pattern once before — in the `SetCurrentText String` value of `Update.Msg`.
+Like `SetCurrentText String`, `SetSelectedKey String` is a **tagged value**. In effect, we are saying that a `SetCurrentText` and `SetSelectedKey` are only valid values of `Msg` union type if they are accompanies by a string.
```

Comment:
Something that helps to explain this a lot on Slack is to talk about how they are constructors for something of the type `Msg`. 

```
type Msg 
  = SetCurrentText String 
  | Reload
```
has two constructors. These can also be called `tags`. Think of a tag as a way of labelling some data, so that it can be grouped together logically with other, related tags. The tag `Reload : Msg` is a value. It is always a tag for the type `Msg`. It takes no arguments. `SetCurrentText` however, does take an argument. It's actually a tagging function - it takes a string, then generates a tag for that string. As a result, it has the type `SetCurrentText : String -> Msg`. 

That way, it's clear that `SetCurrentText` is actually a function - making it trivial to expain how to use it with `onInput :  (String -> msg) -> Attribute msg`, since the types match :)